html/includes/output/query.inc.php

```diff
@@ -43,7 +43,7 @@
                 $rule['query'] = GenSQL($rule['rule']);
             }
             $sql = $rule['query'];
-            $qry = dbFetchRow($sql, array($device_id));
+            $qry[] = dbFetchRow($sql, array($device_id));
```

Comment:
No, $qry is passed later on and needs to be an array, the $sql is a single row lookup but we're in a loop so need to build that $qry array up.